WeatherApp/Controller/LocationWeatherViewController.swift

```diff
@@ -0,0 +1,172 @@
+//
+//  LocationWeatherViewController.swift
+//  WeatherApp
+//
+//  Created by Marcin Wójciak on 17/01/2020.
+//  Copyright © 2020 Marcin Wójciak. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+
+class LocationWeatherViewController: UITableViewController, CLLocationManagerDelegate, WeatherManagerDelegate {
+    
+    let userDefaults = UserDefaults.standard
+    let locationManager = CLLocationManager()
+    var weatherManager = WeatherManager()
+    
+    var userLocations = [String]()
+    var weatherData = [WeatherModel]()
+    
+    override func viewWillAppear(_ animated: Bool) {
+        
+        
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.register(UINib(nibName: "LocationWeatherCell", bundle: nil), forCellReuseIdentifier: "LocationWeatherCell")
+        
+        weatherManager.delegate = self
+        
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers
+        locationManager.delegate = self
+        locationManager.requestLocation()
+        
+        if userLocations.count == 0 {
+            userLocations = userDefaults.stringArray(forKey: "UserLocations") ?? []
+        }
+        print(userLocations)
+        
+        for location in userLocations {
+            weatherManager.fetchWeatherData(for: location)
+        }
+        
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+        
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        
+    }
+    
+    
+    //MARK: - Add User Location
+    
+    @IBAction func addLocationPressed(_ sender: UIBarButtonItem) {
+        
+        var locationName = UITextField()
+        
+        let alert = UIAlertController(title: "Add New User Location", message: "Please type city name", preferredStyle: .alert)
```

Comment:
Wszystkie stringi wykorzystywane w aplikacji polecam przechowywać w pliku `Localizable.strings`. Wygoogluj sobie jak go utworzyć. Dzięki temu, że wszystkie teksty są w jednym miejscu, mamy ułatwioną edycję tekstów aplikacji. A przede wszystkim w bardzo łatwy sposób możeby dodać obsługę kolejnych wersji językowych do aplikacji.

Reply:
Struktura pliku `Localizable.strings` to:
"klucz" = "wartość";

Użycie klucza wygląda tak:
`let text = NSLocalizedString("klucz", comment: "komentarz do tekstu")`
jest to troche boilerplate, dlatego polecam zrobić extension do `String`a - bardzo ułatwia to życie ;)

```
extension String {
    var localized: String {
        return NSLocalizedString(self, comment: "")
    }
}
```
A używane jest tak:
`let text = "klucz".localized`